gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java

```diff
@@ -1161,6 +1188,12 @@ private void validateBuilder(final Builder builder) {
             if (builder.connectionSetupTimeoutMillis < 1)
                 throw new IllegalArgumentException("connectionSetupTimeoutMillis must be greater than zero");
 
+            if (builder.sslHandshakeTimeoutMillis < 0)
+                throw new IllegalArgumentException("sslHandshakeTimeoutMillis must be greater than or equal to zero");
+
+            if (builder.sslHandshakeTimeoutMillis > builder.connectionSetupTimeoutMillis)
+                throw new IllegalArgumentException("connectionSetupTimeoutMillis cannot be less than sslHandshakeTimeoutMillis");
```

Comment:
nit: make exception message logic order the same as if check
```suggestion
                throw new IllegalArgumentException("sslHandshakeTimeoutMillis cannot be greater than connectionSetupTimeoutMillis");
```